subsys/mgmt/fmfu/src/fmfu_mgmt.c

```diff
@@ -26,6 +26,9 @@ LOG_MODULE_REGISTER(mgmt_fmfu, CONFIG_MGMT_FMFU_LOG_LEVEL);
 
 #define FIRMWARE_SHA_LENGTH 32
 
+static fmfu_callback_t callback;
+static bool should_init = true;
```

Comment:
I think the name of this variable is a bit misleading. To me it sounds like a variable protecting against multiple `init` calls being called (like `initialized` variables typically does).

IMO it would be better to describe what state it represents rather than what action should be performed when it is set.

Just a proposal:

```
static bool update_started; /* Is there an ongoing update */

/* ... */ 

static int fmfu_firmware_upload(struct mgmt_ctxt *ctx)
{
	static bool bootloader = true;
	if (!update_started) {
		callback(FMFU_EVT_INIT);
		update_started = true;
	}
```

Also it would be really good if we could keep this variable inside the scope of the `fmfu_firmware_upload`. This might not be possible, but I would make an effort to see if its possible.

Reply:
sure calling it `update_started` sounds good